deps/wazuh_testing/wazuh_testing/modules/fim/event_monitor.py

```diff
@@ -58,3 +66,72 @@ def callback_detect_event(line):
             return json_event
     except (json.JSONDecodeError, AttributeError, KeyError) as e:
         logger.warning(f"Couldn't load a log line into json object. Reason {e}")
+
+
+def callback_detect_end_scan(line):
+    """ Callback that detects if a line in a log is a scan_end event
+    Args:
+        line (String): string line to be checked by callback. Passed in by File_Monitor.
+    """

```

Comment:
I prefer not to use a specific variable as `scan_end` in the description. 

The parameter should be more representative like `event`, not `line`.

I don't see that adding `Passed in by File_Monitor` clarifies something.

Reply:
The line name has been inherited and matches the rest of the callbacks in fim.py. It is used to keep consistency.

Reply:
Regarding the requested test cases:
- The first 3 cases, with default values will not be added, because they would be too long and there is no actual way to validate the default values from the configuration:
   - `report_changes` default value is 30, so it would take a test an `interval` of 30+ for it to fire an interval doubling, and then again 60 seconds until it goes back to normal values. But since there is no log with the configuration value, we have no actual way of knowing this is the value, we would be just assuming.
   - `max_interval` has a default value of 3600, so in order for a test to exceed the default, it would take an 1 hour of uninterrupted sync. This is not a useful test. We are checking the functionality of `max_interval` in the other test cases with lower values.
- The fourth case is already covered in the test case `Invalid max_interval value (Lower than interval)`
- The last case will be implemented by adding few files so that the interval is not doubled and the sync happens in less than the configured max_interval.